src/plots/plots.js

```diff
@@ -1813,6 +1813,10 @@ plots.transition = function(gd, data, layout, traces, frameOpts, transitionOpts)
         // of essentially the whole supplyDefaults step, so that it seems sensible to just use
         // supplyDefaults even though it's heavier than would otherwise be desired for
         // transitions:
+
+        // first delete calcdata so supplyDefaults knows a calc step is coming
+        delete gd.calcdata;
```

Comment:
There's some [logic](https://github.com/plotly/plotly.js/blob/6b7c9d7aeba575c3ff5a35a03e111c051e001ed7/src/plots/plots.js#L2026) in `Plots.doCalcdata` involving old versus new calcdata that might be obsolete now.